src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -703,6 +703,62 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
         billingPostcode: this.giftAidGroup.value.homePostcode,
       });
     }
+
+    if (
+      event.selectedIndex > this.paymentStepIndex &&
+      this.donor &&
+      this.donation
+    ) {
+      this.updateNewPersonInBackend(this.donation);
+    }
+  }
+
+  /**
+   * Updates the name and email address of the new person in the backend. Required at this stage for
+   * the email address in particular so that there will be a ready to use email verification token
+   * for when the donation is confirmed.
+   */
+  private updateNewPersonInBackend(donation: Donation) {
+    const idAndJWT = this.identityService.getIdAndJWT();
+    if (idAndJWT) {
+      let person = this.buildPersonFromDonation(donation);
+      person.id = idAndJWT.id;
+
+      // Try to patch the person only if they're not already a finalised donor account,
+      // e.g. they could have set a password then reloaded this page.
+      if (!this.identityService.isTokenForFinalisedUser(idAndJWT.jwt)) {
+        this.identityService.update(person)
+          .subscribe({
+            next: person => {
+              this.donor = person;
+            },
+            error: (error: HttpErrorResponse) => {
+              // For now we probably don't really need to inform donors if we didn't patch their Person data, and just won't ask them to
+              // set a password if the first step failed. We'll want to monitor Analytics for any patterns suggesting a problem in the logic though.
+              this.matomoTracker.trackEvent('identity_error', 'person_core_data_update_failed', `${error.status}: ${error.message}`);
+            },
+          });
+      } // End token-not-finalised condition.
+    } // End ID-feature-enabled condition.
+  }
+
+  private buildPersonFromDonation(donation: Donation): Person {
+    const idAndJWT = this.identityService.getIdAndJWT();
+
+    let person: Person = {
+      id: idAndJWT!.id,
+      email_address: donation.emailAddress,
+      first_name: donation.firstName,
+      last_name: donation.lastName,
+    };
+
+    if (donation.giftAid) {
+      person.home_address_line_1 = donation.homeAddress;
+      person.home_postcode = donation.homePostcode === OVERSEAS ? undefined : donation.homePostcode;
+      person.home_country_code = donation.homePostcode === OVERSEAS ? OVERSEAS : 'GB';
+    }
+
+    return person;
   }
```

Comment:
functions moved mostly unchanged from thanks component

Reply:
possibly small risk that if someone reloads the site before going to thanks and switches from a version before this PR to a version after they won't be able to save their password, but I think that's an acceptable risk if we don't deploy at a busy time.